cylc/sphinx_ext/cylc_lang/autodocumenters.py

```diff
@@ -346,3 +367,314 @@ def run(self):
+            if not info.get('title', ''):
+                meta['runtime'][namespace]['title'] = namespace
+
+        if poproot:
+            meta['runtime'].pop('root')
+
+        return meta
+
+    @staticmethod
+    def workflow_to_node(meta, src):
+        """Convert workflow metadata to list of strings for use as a node.
+
+        """
+        ret = []
+
+        # Handle the workflow config metadata:
+        workflow_title = meta.get('workflow', {}).get('title', src)
+        workflow_desc = meta.get('workflow', {}).get(
+            'description', '').split('\n')
+
+        ret += [f'.. cylc:conf:: {workflow_title}', '']
+        ret += [f'   {t.strip()}' for t in workflow_desc] + ['']
+
+        # Handle the runtime config metadata:
+        ret += [f'{INDENT}Runtime', f'{INDENT}=======', '']
+
+        for taskmeta in meta['runtime'].values():
+            indent = INDENT
+            title = taskmeta["title"]
+            ret += [f'{indent}.. cylc:conf:: {title}', '']
+            indent += INDENT
+
+            url = taskmeta.get('url', '')
+            if url:
+                ret += [
+                    f'{indent}.. seealso::',
+                    '',
+                    f'{indent + INDENT}{url}',
+                    ''
+                ]
+
+            desc = taskmeta.get('description', '').replace('\n', ' ')
+            ret += [f'{indent}{desc}', '']
+
+        ret += CylcMetadataDirective.meta_to_def_list(
+            metadata=taskmeta.items(),
+            lowest_indent=1
+        )
+
+        # Handy for debugging:
+        # [print(f'{i + 1:03}|{j}') for i, j in enumerate(ret)]
+
+        return ret
+
+    @staticmethod
+    def check_subproc_output(sub):
+        """Check subprocess outputs - catch failure.
+        """
+        if sub.returncode:
+            # Very specifically handle the case where the correct
+            # version of Cylc isn't installed:
+            if 'no such option: --json' in sub.stderr.decode():
+                msg = (
+                    'Requires cylc config --json, not available'
+                    ' for this version of Cylc')
+                raise DependencyError(msg)
+            # Handle any and all other errors in the subprocess:
+            else:
+                msg = 'Cylc config metadata failed with: \n'
+                msg += '\n'.join(
+                    i.strip("\n") for i in sub.stderr.decode().split('\n'))
+                raise Exception(msg)
+
+    @staticmethod
+    def meta_to_def_list(metadata, lowest_indent):
+        result = []
+        other_meta_items = False
+        for key, value in metadata:
+            if key in ['title', 'description', 'url']:
+                continue
+            if other_meta_items is False:
+                result += [
+                    f'{INDENT * lowest_indent}Other metadata',
+                    f'{INDENT * lowest_indent}^^^^^^^^^^^^^^', '']
```

Comment:
Use a rubric directive here.

Note, you can't know what the title level is because you don't know what document this directive is going to be used in so using titles like this might not work.